src/config_parser_utils/merge_configuration_files.py

```diff
@@ -15,59 +19,105 @@
 """
 import configparser
 import logging
+from logging.handlers import SysLogHandler
+import shutil
 import sys
 import os
+import re
+
+CFG_LINE_RGX = r"^(\S+)\s*=\s*(.*)$"
+
+def setup_logger():
+    """Configures and returns a logger that sends logs to syslog."""
+    logger = logging.getLogger('upgrade')
+    logger.setLevel(logging.DEBUG)
+
+    # Create a syslog handler for the local syslog daemon
+    syslog_handler = SysLogHandler(address=('localhost', 514))
```

Comment:
what if the host is listening to a different port for the syslog?
I recommend you to use the local default unix socket path

```suggestion
    syslog_handler = SysLogHandler(address='/dev/log')
```

Reply:
what if the host is not ubuntu but a red hat? will it still work?

Reply:
yes

Reply:
When I firstly tried to run it with address='/dev/log' I got FileNotFoundError because the container does not share '/dev/log' with the host

Reply:
@Miryam-Schwartz , following our conversation, we discovered that the plugin manager automatically forwards all stdout from upgrade.sh to the syslog. This means you don’t need to handle syslog logging within your script. The plugin manager will take care of forwarding the output to the syslog.

please check it within your script.